### PR TITLE
 [VarDumper] Add dq() function for SQL query debugging

### PR DESCRIPTION
**Pull request by Jewhurst**
This adds a new debugging helper function that formats and dumps SQL queries with their bindings for easier debugging.

| Q             | A
| ------------- | ---
| Branch?       | 7.3
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Issues        | n/a
| License       | MIT

### Description
Adds a new `dq()` (dump query) function that helps debug SQL queries by showing the complete query with bound parameters interpolated into the SQL string.

### CODE REVIEW

File: src/Symfony/Component/VarDumper/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,11 @@
 CHANGELOG
 =========
 
+7.2.1
```
Comment (derrabus) [inline src/Symfony/Component/VarDumper/CHANGELOG.md:4]:
```suggestion
7.3
```

File: src/Symfony/Component/VarDumper/Resources/functions/dump.php [hunk 2]
```diff
@@ -66,3 +66,35 @@ function dd(mixed ...$vars): never
...
+     *
+     * @param mixed $query The query builder instance
+     */
+    function dq(mixed $query): never
+    {
+        if (!\in_array(\PHP_SAPI, ['cli', 'phpdbg', 'embed'], true) && !headers_sent()) {
+            header('HTTP/1.1 500 Internal Server Error');
+        }
+
+        if (!method_exists($query, 'toSql') || !method_exists($query, 'getBindings')) {
```
Comment (derrabus) [inline src/Symfony/Component/VarDumper/Resources/functions/dump.php:82]: Can we avoid the duck typing here? What object exactly are you expecting?

**Review by derrabus — CHANGES_REQUESTED** [review]
Tests are missing.

**Comment by lyrixx** [comment]
I would use a caster here with a virtual property

**Comment by dsentker** [comment]
I don't know if we need a separate function for such a simple task. You can also get an executable query in the Symfony Profiler.

**Comment by derrabus** [comment]
Indeed, the `dump()` function should be good enough for the task. And if we're not happy with its output, a dedicated caster would do the trick.

Let's close this PR.

**Pull request closed without merging**